cmake/recipes/external/cgal.cmake

```diff
@@ -34,8 +34,10 @@ function(cgal_import_target)
     include(boost)
 
     ignore_package(GMP 5.0.1)
-    set(GMP_INCLUDE_DIR "")
+    set(GMP_INCLUDE_DIR ${gmp_INCLUDE_DIR})
```

Comment:
This is not necessary, since our target `gmp::gmp` should carry the include dir already.

Reply:
I think CGAL's cmake is looking at `GMPXX_INCLUDE_DIR` so it needs to be explicitly set.

Reply:
Confirmed, if `GMPXX_INCLUDE_DIR` is set to `""` then CGAL determines that `CGAL_WITH_GMPXX:BOOL=OFF`. By setting `GMPXX_INCLUDE_DIR` explicitly then we keep `CGAL_WITH_GMPXX:BOOL=ON`.

Reply:
If you add `ignore_package(GMPXX 5.0.1)`, you can probably leave `GMP_INCLUDE_DIR` and `GMPXX_INCLUDE_DIR` set to `""`, since the fake `GMPXXConfig.cmake` will set `GMPXX_FOUND` to `ON`, and `CGAL_WITH_GMPXX` should subsequently be set to `ON`.

Reply:
ah, got it. Didn't realize these were related. I pushed the change.

Reply:
augh... that broke Windows. Is there anything wrong with using the previous version with GMPXX_INCLUDE_DIR explicitly set? At least windows was building...

Reply:
alecjacobson/gmpxx-include-dir

Reply:
augh, that broke windows. Anything wrong with explicitly setting GMP_INCLUDE_DIR and GMPXX_INCLUDE_DIR ? That was at least working on windows...

Reply:
hmm, actually seems that the ignore_package is causing trouble on windows...

Reply:
Well that's because gmpxx is **not** supported on Windows. So actually we might want to disable GMPXX directly by using [CMAKE_DISABLE_FIND_PACKAGE_<PackageName>](https://cmake.org/cmake/help/v3.16/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html#variable:CMAKE_DISABLE_FIND_PACKAGE_%3CPackageName%3E). I.e. don't add `ignore_package(GMPXX ...)`, do not set `GMPXX_INCLUDE_DIR` or `GMPXX_LIBRARIES`, but add the following line before including CGAL:
```cmake
set(CMAKE_DISABLE_FIND_PACKAGE_GMPXX ON)
```

Reply:
so, when the windows CI was succeeding was CGAL just not enabling gmpxx? Why this extra code to explicitly disable it then?

Reply:
Nope. GMPXX is an optional dependency of CGAL, so if CGAL doesn't find GMPXX it continues with its own internal wrapper around gmp objects. If we force CGAL into pretending we found GMPXX (via our `ignore_package(GMPXX)`) then it will fail when it actually tries to include `gmpxx.h`.

Reply:
Why would the CI succeed then?

Reply:
We're back to the state of the original PR which passes CI. If there's a more elegant way to handle GMPXX that maintains passing for Windows then let's merge it in a new PR. I'll merge this for now.

Reply:
What do you mean by why would the CI succeed? With your changes you are not preventing CGAL's FindGMPXX from executing. But because you set the `GMPXX_INCLUDE_DIR` variable in advance, CGAL is using those locations. On Windows, it will still go through this [find_path()](https://github.com/CGAL/cgal/blob/a8e87526348d2d1faa2ea893747314853ee6868d/Installation/cmake/modules/FindGMPXX.cmake#L20) command and silently fail to find GMPXX. On Linux/macOS, I am not sure whether it find them or not.

In the end I suppose your setup kinda works, though you may think you're using GMPXX on Windows when you are not.

Reply:
Right. In the merge version `GMPXX_INCLUDE_DIR` is set to point to the gmp build for all systems, and I guess on windows cgal will realize that gmpxx should be off so it's ignored. Using the `ignore_package` caused windows build to fail out of the box. If there's a solution that avoids pretending that gmpxx is really there then we should merge it, but I'm not sure what that is and I end up pulling my hair out trying to debug the windows build.

Reply:
Well the solution is simple:
```cmake
if(WIN32)
    set(CMAKE_DISABLE_FIND_PACKAGE_GMPXX ON)
else()
    ignore_package(GMPXX 5.0.1)
    set(GMPXX_INCLUDE_DIR "")
    set(GMPXX_LIBRARIES ${GMP_LIBRARIES})
endif()
```

The end result should be basically the same though...

Reply:
😊 Want to PR this and merge?